src/fqm.ts

```diff
@@ -42,39 +42,54 @@ export const calculateMeasureReports = async (
 };
 
 /**
- * Populates the _type parameter used in a bulk data export request. Retrieves the data
- * requirements for the given measure bundle using the fqm-execution API function. Extracts
- * the resource types from the data requirements. Throws error if data requirements are not
- * defined on the results of the API function.
+ * Populates the _type/_typeFilter parameters used in a bulk data export request. Retrieves the data
+ * requirements for the given measure bundle using the fqm-execution API function to use for
+ * _type and _typeFilter query construction. Throws error if data requirements are not defined on the results
+ * of the API function.
  * @param measureBundle FHIR measure bundle
+ * @param autoType boolean specifying whether we want to auto-generate _type
+ * @param autoTypeFilter boolean specifying whether we want to auto-generate _typeFilter
  * @param options fqm-execution calculation options
  */
-export const retrieveTypeFromMeasureBundle = async (
+export const retrieveParamsFromMeasureBundle = async (
   measureBundle: fhir4.Bundle,
+  autoType: boolean,
+  autoTypeFilter: boolean,
   options: CalculatorTypes.CalculationOptions = {}
 ) => {
   const dr = await Calculator.calculateDataRequirements(measureBundle, options);
   if (!dr.results.dataRequirement) {
     throw new Error('Data requirements array is not defined for the Library. Aborting $export request.');
   }
-  return constructTypeQueryFromRequirements(dr.results.dataRequirement);
+  return constructParamsFromRequirements(dr.results.dataRequirement, autoType, autoTypeFilter);
 };
 
 /**
- * Constructs the _type parameter used in a bulk data export request by parsing the
- * data requirements for the measure bundle.
+ * Constructs the _type and _typeFilter parameters used in a bulk data export request by parsing the
+ * data requirements from the measure bundle.
  * @param dataRequirements data requirements retrieved from measure bundle
+ * @param autoType boolean specifying whether we want to auto-generate _type
+ * @param autoTypeFilter boolean specifying whether we want to auto-generate _typeFilter
  */
-export const constructTypeQueryFromRequirements = (dataRequirements: fhir4.DataRequirement[]) => {
+export const constructParamsFromRequirements = (
+  dataRequirements: fhir4.DataRequirement[],
+  autoType: boolean,
+  autoTypeFilter: boolean
+) => {
+  const queries: { type: string; params: Record<string, string> }[] = [];
   const types: string[] = [];
-
   dataRequirements.forEach((dr) => {
     if (dr.type) {
       types.push(dr.type);
+      const query: { type: string; params: Record<string, string> } = { type: dr.type, params: {} };
+      if (dr?.codeFilter?.[0]?.valueSet) {
+        const key = `${dr?.codeFilter?.[0].path}:in`;
+        key && (query.params[key] = dr.codeFilter[0].valueSet);
+      }
+      queries.push(query);
```

Comment:
This code is also limited to code filters.  Did you also want to include some support for dateFilters?  For example

![Screenshot 2023-08-17 at 1 01 30 PM](https://github.com/bulk-dqm/bulk-data-export-client/assets/8173551/9fa2c58a-a6a3-4664-bede-a785c8102b6f)

Reply:
I think overall it would be beneficial to add date filters, but right now the Abacus export server does not support filtering by date filters and would throw an error. I'll try implementing support for this and see if the date filter queries get ignored when I use the `Prefer: handling=lenient` header. If they do get ignored, then I would say date filter support is in scope of this work. If not, maybe we should make a backlog task to support date filter queries in the future once we have support for it in the export server.